src/ballet/gossip/fd_gossip_validation.h

```diff
@@ -0,0 +1,56 @@
+
+#define FD_GOSSIP_MTU                                  (1232UL)
+#define FD_GOSSIP_CRDS_MAX_WALLCLOCK                   (1000000000000000UL)
+#define FD_GOSSIP_CRDS_MAX_VOTES                       (32UL)
+#define FD_GOSSIP_CRDS_MAX_EPOCH_SLOTS                 (255UL)
+#define FD_GOSSIP_CRDS_MAX_DUPLICATE_SHREDS            (512UL)
+#define FD_GOSSIP_CRDS_MAX_SLOT                        (1000000000000000UL)
+#define FD_GOSSIP_CRDS_MAX_SLOTS_PER_ENTRY             (2048*8)
+
+#define CHECK_WALLCLOCK( wallclock )  do {                                 \
+      if( FD_UNLIKELY( wallclock>FD_GOSSIP_CRDS_MAX_WALLCLOCK ) ) {        \
+        FD_LOG_WARNING(( "wallclock exceeds max value" ));                 \
+        return 0;                                                          \
+      }                                                                    \
+    } while(0)
+
+#define CHECK_LOWEST_SLOT_INDEX( index )  do {                             \
+      if( FD_UNLIKELY( index>=1 ) ) {                                      \
+        FD_LOG_WARNING(( "invalid slot_index value" ));                    \
+        return 0;                                                          \
+      }                                                                    \
+    } while(0)
+
+#define CHECK_VOTE_INDEX( index )  do {                                    \
+      if( FD_UNLIKELY( index>=FD_GOSSIP_CRDS_MAX_VOTES ) ) {               \
+        FD_LOG_WARNING(( "invalid vote index value" ));                    \
+        return 0;                                                          \
+      }                                                                    \
+    } while(0)
+
+#define CHECK_EPOCH_SLOTS_INDEX( index )  do {                             \
+      if( FD_UNLIKELY( index>=FD_GOSSIP_CRDS_MAX_EPOCH_SLOTS ) ) {         \
+        FD_LOG_WARNING(( "invalid epoch slots index value" ));             \
+        return 0;                                                          \
+      }                                                                    \
+    } while(0)
+
+#define CHECK_DUPLICATE_SHRED_INDEX( index )  do {                         \
+      if( FD_UNLIKELY( index>=FD_GOSSIP_CRDS_MAX_DUPLICATE_SHREDS ) ) {    \
+        FD_LOG_WARNING(( "invalid epoch slots index value" ));             \
+        return 0;                                                          \
+      }                                                                    \
+    } while(0)
+
+#define CHECK_SLOT( slot ) do {                                            \
+     if( FD_UNLIKELY( slot>=FD_GOSSIP_CRDS_MAX_SLOT ) ) {                  \
+       FD_LOG_WARNING(( "invalid slot" ));                                 \
+       return 0;                                                           \
+     }                                                                     \
+} while(0) 
+
+#endif
```

Comment:
mkdir: created directory 'build/linux/clang/x86_64/include/ballet/txn/'
src/ballet/gossip/fd_gossip_validation.h:56:7: error: no newline at end of file [-Werror,-Wnewline-eof]
#endif
      ^
In file included from src/ballet/gossip/fd_gossip_msg_decode.c:6:
src/ballet/gossip/fd_gossip_vector_utils.h:23:7: error: no newline at end of file [-Werror,-Wnewline-eof]
#endif
      ^
mkdir -pv build/linux/clang/x86_64/include/ballet/gossip/ && \